src/iqm/qiskit_iqm/iqm_provider.py

```diff
@@ -82,9 +83,44 @@ def max_circuits(self, value: Optional[int]) -> None:
         self._max_circuits = value
 
     def run(self, run_input: Union[QuantumCircuit, list[QuantumCircuit]], **options) -> IQMJob:
+        run_request = self.create_run_request(run_input, transpile_circuits=False, **options)
+        job_id = self.client.submit_run_request(run_request)
+        job = IQMJob(self, str(job_id), shots=run_request.shots)
+        job.circuit_metadata = [c.metadata for c in run_request.circuits]
+        return job
+
+    def create_run_request(
+        self, run_input: Union[QuantumCircuit, list[QuantumCircuit]], *, transpile_circuits=True, **options
+    ) -> RunRequest:
+        """Creates a run request without submitting it for execution.
+
+        This can be used to check what would be submitted for execution by an equivalent call to :meth:`run` or
+        Qiskit's :func:`execute` function.
+
+        By default, transpiles the circuits in the same way as Qiskit's :func:`execute` function. Set
+        ``transpile_circuits=False`` to get the same behavior as calling :meth:`run` directly.
+
+        NOTE: because transpilation in Qiskit can be stochastic, the returned run request can be different than what
+        would be sent by a subsequent call to :func:`execute`. To avoid any differences, either set the same value for
+        ``seed_transpiler`` in both calls, or call :meth:`~iqm.iqm_client.IQMClient.submit_run_request` with the
+        returned run request instead of calling :func:`execute`.
+
+        Args:
+            run_input: same as ``run_input`` for :meth:`run`
+            transpile_circuits: If ``True``, transpile circuits in the same way as Qiskit's :func:`execute`.
+                Otherwise don't transpile circuits.
+            options: transpilation options similar to Qiskit's :func:`execute`, and other options similar to :meth:`run`
+
+        Returns:
+            the created run request object
+        """
         if self.client is None:
             raise RuntimeError('Session to IQM client has been closed.')
 
+        if transpile_circuits:
+            # TODO: remove this when moving to Qiskit 1.0 which removes the execute function
+            run_input = self._transpile_circuits(run_input, **options)
+
```

Comment:
The original idea of adding transpilation here was that one can inspect the run request that would be submitted by `execute` by doing a single call with same parameters, without actually submitting the request (which would not be possible when calling `execute`). And the reason for supporting similar behavior as `execute` instead of just direct `backend.run` was because our documentation generally suggested using `execute`.

But you are right that this should probably be removed already, since our move to Qiskit v1.0 seems to be coming sooner than I thought when starting to implement this. It would also indeed cause unnecessary backwards compatibility issue.

Removed all the transpilation related things from this now.

Reply:
`execute` is just a wrapper around `transpile` and `backend.run()`, so you can pass the same device specific arguments for `run` to `execute`. (they are passed through the `run_config` kwargs in `execute`.

Reply:
Yes, that's clear, it's just that this was originally supposed to be a single call that takes the "transpilation params" + "run params" exactly like `execute`, without having to do transpilation in  a separate call first. But now it doesn't matter since the transpilation part is removed in anticipation of Qiskit 1.0.